File: roman-to-integer/romanToInt.js
/**
 * @param {string} s
 * @return {number}
 */
var romanToInt = function (s) {
  let result = 0;
  s = s + " ";
  for (let i = 0; i < s.length; i++) {
    const next = s[i + 1];
    switch (s[i]) {
      case "I":
        if (!["V", "X"].includes(next)) {
          result++;
        } else {
          result--;
        }
        break;
      case "V":
        result += 5;
        break;
      case "X":
        if (!["L", "C"].includes(next)) {
          result += 10;
        } else {
          result -= 10;
        }
        break;
      case "L":
        result += 50;
        break;
      case "C":
        if (!["D", "M"].includes(next)) {
          result += 100;
        } else {
          result -= 100;
        }
        break;
      case "D":
        result += 500;
        break;
      case "M":
        result += 1000;
        break;
    }
  }

  return result;
};

File: roman-to-integer/RomanToInt.cs
public class Solution
{
    public int RomanToInt(string s)
    {
        int result = 0;
        Dictionary<char, int> map = new() { { 'I', 1 }, { 'V', 5 }, { 'X', 10 }, { 'L', 50 }, { 'C', 100 }, { 'D', 500 }, { 'M', 1000 } };

        for (int i = 0; i < s.Length - 1; i++)
        {
            var next = s[i + 1];

            if (map[s[i]] < map[next])
                result -= map[s[i]];
            else
                result += map[s[i]];
        }

        result += map[s[s.Length - 1]];

        return result;
    }
}